### PR TITLE
executor: better error message for `ADMIN CHECK INDEX`

### LINKED ISSUE #37323: *: better error message for `ADMIN CHECK/RECOVER/CLEANUP INDEX`
## Enhancement

When a table is defined with clustered index, the 'primary key' does not exist actually(primary key = record), so for syntax like `ADMIN CHECK INDEX xxx `primary``, a better error message would be better:

Current:
```
tidb> create table t(a int key clustered);
tidb> admin recover index t `primary`;
ERROR 1105 (HY000): index `primary` is not found in table `t`
```

Not every user knows the definition of "clustered index", we could report another message like:

```
ERROR 1105 (HY000): index `primary` is not defined in table `t` when the clustered index is defined.
```

### PR DESCRIPTION
**Pull request by taofengliu**
### What problem does this PR solve?

Issue Number: close #37323 

Problem Summary:

### What is changed and how it works?
Maybe the better error message should be "index `primary` is not defined in table `%v` when primary key is a single integer column and clustered" ?
### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by taofengliu** [comment]
@bb7133 Could you please review this PR?

**Review by CbcWestwolf — COMMENTED** [review]
Thanks for your contribution! :-)
Could you please add more unit tests in proper functions in https://github.com/pingcap/tidb/blob/master/executor/admin_test.go? It is better if the added test cases cover all potential introduced paths in this PR, which are some illegal `admin check`/`admin recover`/`admin cleanup` statements.

**Comment by taofengliu** [comment]
> Thanks for your contribution! :-) Could you please add more unit tests in proper functions in https://github.com/pingcap/tidb/blob/master/executor/admin_test.go? It is better if the added test cases cover all potential introduced paths in this PR, which are some illegal `admin check`/`admin recover`/`admin cleanup` statements.

OK, I will add tests.

**Review by CbcWestwolf — APPROVED** [review]

**Comment by taofengliu** [comment]
Thanks so much for reviewing this PR! @CbcWestwolf 
Because first-time contributors need a maintainer to approve running workflows, could you please address this then this PR can be merged.

**Comment by CbcWestwolf** [comment]
> could you please address this then this PR can be merged.

Sure! I'll look for someone for help.

**Comment by CbcWestwolf** [comment]
/cc tangenta
PTAL

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -1790,7 +1790,10 @@ func (b *PlanBuilder) buildAdminCheckTable(ctx context.Context, as *ast.AdminStm
 			}
 		}
 		if idx == nil {
-			return nil, errors.Errorf("index %s do not exist", as.Index)
+			if idxName == "primary" && tbl.Meta().PKIsHandle {
+				return nil, errors.Errorf("index `primary` is not defined when primary key is a single integer column and clustered")
+			}
+			return nil, errors.Errorf("index %s does not exist", as.Index)
```
Comment (tangenta) [inline planner/core/planbuilder.go:1796]:
```suggestion
			if idxName == "primary" && tbl.Meta().PKIsHandle {
				return nil, errors.Errorf("index `primary` is not defined when primary key is a single integer column and clustered")
			}
			return nil, errors.Errorf("index %s does not exist", as.Index)
```

How about changing the error message to "secondary index %s does not exist"? I think it is much clear than "index `primary` is not defined when primary key is a single integer column and clustered".

Comment (taofengliu) [reply planner/core/planbuilder.go:1796]: Do you mean ADMIN CHECK/RECOVER/CLEANUP INDEX is only used for secondary index ?

Comment (tangenta) [reply planner/core/planbuilder.go:1796]: @taofengliu `ADMIN * INDEX` can be also used for non-clustered primary key(I usually consider it as a special secondary index).

**Review by tangenta — APPROVED** [review]

**Pull request merged**